plugins/input/raster/raster_featureset.cpp

```diff
@@ -89,33 +89,36 @@ feature_ptr raster_featureset<LookupPolicy>::next()
                     box2d<double> intersect = bbox_.intersect(curIter_->envelope());
                     box2d<double> ext = t.forward(intersect);
                     box2d<double> rem = policy_.transform(ext);
-                    if (ext.width() > 0.5 && ext.height() > 0.5 )
-                    {
-                        // select minimum raster containing whole ext
-                        int x_off = static_cast<int>(std::floor(ext.minx()));
-                        int y_off = static_cast<int>(std::floor(ext.miny()));
-                        int end_x = static_cast<int>(std::ceil(ext.maxx()));
-                        int end_y = static_cast<int>(std::ceil(ext.maxy()));
-
-                        // clip to available data
-                        if (x_off < 0) x_off = 0;
-                        if (y_off < 0) y_off = 0;
-                        if (end_x > image_width)  end_x = image_width;
-                        if (end_y > image_height) end_y = image_height;
-
-                        int width = end_x - x_off;
-                        int height = end_y - y_off;
-
-                        // calculate actual box2d of returned raster
-                        box2d<double> feature_raster_extent(rem.minx() + x_off,
-                                                            rem.miny() + y_off,
-                                                            rem.maxx() + x_off + width,
-                                                            rem.maxy() + y_off + height);
-                        intersect = t.backward(feature_raster_extent);
-                        mapnik::image_any data = reader->read(x_off, y_off, width, height);
-                        mapnik::raster_ptr raster = std::make_shared<mapnik::raster>(intersect, std::move(data), 1.0);
-                        feature->set_raster(raster);
+                    // select minimum raster containing whole ext
+                    int x_off = static_cast<int>(std::floor(ext.minx()));
+                    int y_off = static_cast<int>(std::floor(ext.miny()));
+                    int end_x = static_cast<int>(std::ceil(ext.maxx()));
+                    int end_y = static_cast<int>(std::ceil(ext.maxy()));
+
```

Comment:
No I did not, it is something that I still need to dig into -- I probably should just draw it out that always helps the most.